packages/social-previews/package.json

```diff
@@ -0,0 +1,37 @@
+{
```

Comment:
Looks like this is missing all package dependencies. I think these are the deps, but I may be missing some:

```
	"dependencies": {
		"i18n-calypso": "^5.0.0",
		"lodash": "^4.17.15",
		"prop-types": "^15.7.2"
	},
	"peerDependencies": {
		"react": "^16.12.0"
	}
```

Reply:
I was about to comment the same but since the build takes place in Calypso which already has these deps, are they really needed? I mean, in that case shouldn't also `calypso-build` be a dependency since it contains the `transpile` ?

Reply:
Not all dependencies are in calypso (as in, the root package). For example, `prop-types` is missing. This works by coincidence, because somebody else depends on `prop-types` and it gets hoisted to the root package.

And if this package is meant to be published, nobody will be able to use it if the dependencies are missing, including `calypso-build` that as you said it should be a devDependency as well, I missed that one.

Reply:
Ah, cool! In that case, I totally agree 😄 ! Thanks for the info!

Reply:
Thanks for flagging this @scinos. 

It might just be me but this wasn't super clear to me from reading [the documentation](https://github.com/Automattic/wp-calypso/blob/master/docs/monorepo.md).

So I have to go through every file, find the deps and add them. Then I have to make sure I know about all the `devDeps` and add those too? Is there a way to validate that the package can be consumed independently of the Calypso monorepo (ie: it lists all it's deps correctly)?

Reply:
Yeah, it is a lot of manual work. Package extraction is not easy :(

I got the list by running `npx @yarnpkg/doctor` inside `packages/social-previews`. Then manually compose the JSON and get the versions from the root `package.json`. From there, you have to judge if a dep is a regular dependency or devDependency.

Reply:
Note I had to run `npx @yarnpkg/doctor packages/social-previews` from the root dir.

Reply:
Also had to remember to build the package before running the doctor command. So in full:

`yarn workspace @automattic/social-previews run prepare && npx @yarnpkg/doctor packages/social-previews`.

Reply:
@scinos I've ended up with this result:

![Screen Shot 2020-07-22 at 12 22 34](https://user-images.githubusercontent.com/444434/88170776-2593a000-cc16-11ea-9adf-9aaaec35e376.png)

Does the `package.json` look better to you now?

Reply:
Yeah, much better now!!
Thanks for fixing it.